src/other/Table.js

```diff
@@ -557,8 +603,41 @@
         HTMLWidget.prototype.exit.apply(this, arguments);
     };
         
+    Table.prototype.setColumnWidths = function() {
+        var context = this;
+        var tableMarginHeight = this.fixedHeader() ? this.thead.node().offsetHeight : 0;
+
+        var totalWidth = 0;
+        var tds = this.table.select("tbody").selectAll("tr").selectAll("td");
+        var th = this.thead.selectAll("th").data(this.showHeader() ? this.columns() : [], function (d) { return d; });
+        th.each(function(d, i) {
+            var thwidth = this.offsetWidth;
+            var tdwidth = tds.empty() || !tds[0][i] ? 0 : tds[0][i].offsetWidth;
```

Comment:
@GordonSmith If a td is empty, I set the width to 0.  The value obtained here is then compared to the width of the corresponding header cell (thWidth), and the larger of the two values is used as the width for both.